crates/forge_analyzer/src/definitions.rs

```diff
@@ -226,9 +233,92 @@ pub fn run_resolver(
         module.visit_with(&mut collector);
     }
 
+    // This loop iterates through env's bodies and recreates an updated set of bodies to satisfy SSA form of IR dump.
+    let mut vars_map: HashMap<VarId, (VarKind, Option<VarId>, bool)> = HashMap::new();
+    for (func_id, body) in environment.bodies_mut().enumerate() {
+        // Builds the variables map to track existing all variables, store their VarId, and check if they have been assigned once (by SSA).
+        for var_id in body.vars.keys() {
```

Comment:
Actually wait, do we need `Option<VarId>` if we can just insert the `VarId` into the map if its only been assigned? We can also probably use a separate `TiVec` of bools, since that has to be present for every `VarId`.

You should be able to get the `VarKind` by directly indexing `body.vars` as well.